cfep-09.md

```diff
@@ -0,0 +1,280 @@
+
+Names don't sort well and as such are a bit problematic.  However we can impose a new name by 
+stating what the ordering of names are.  This allows us to effectively treat a name migration exactly the the same as
+a version migration because we have an ordering that we can impose on the system.
+
+<table>
+<tr>
+<td>
+
+```yaml
+# Existing pins
+c_compiler:
+  - toolchain_c
+```
+</td>
+<td> with </td>
+<td>
+
+```yaml
+# migrator
+__migrator:
+  kind: version
+  ordering:
+    c_compiler:
+      - toolchain_c
+      - gcc
+c_compiler:
+  - gcc
+```
+</td>
+<td> -> </td>
+<td>
+
+```yaml
+# resulting pins
+c_compiler:
+  - gcc
+```
+</td>
+</tr>
+</table>
+
+
+#### 4. Key removal
+
+Some times we want to perform a migrator where we delete a given value and don't wish to substitute it with a new one.
+A deletion migration is designed purely to remove values
+
+<table>
+<tr>
+<td>
+
+```yaml
+# Existing pins
+ruamel_yaml:
+  - 1.40
+numpy:
+  - 1.14
+```
+</td>
+<td> with </td>
+<td>
+
+```yaml
+# migrator
+__migrator:
+  kind: deletion
+ruamel_yaml:
+  - 1.40
+```
+</td>
+<td> -> </td>
+<td>
+
+```yaml
+# resulting pins
+numpy:
+  1.14
+```
+</td>
+</tr>
+</table>
+
+Key removals followed by additions are a viable way to perform a migration to a lower version.  We have never done this really.
+
+## Rationale
+
+### Multiple migrations of the same package:
+A feedstock can get multiple migration PRs at once, it is up to the feedstock maintainers to choose which PRs to merge in what order.
+
+
+## Backwards Compatibility
+This is fully backwards compatible with the current ecosystem.
+
+## Alternatives
+
+1. Earlier versions of this proposal did not direct version the global pinnings files in a time ordered fashion. Instead they simply always preferred the local pinnings over the global ones. However, this procedure fails in the case of a package adding a new dependence with a pin and it having an outdated local pinnings file. By having a version stored in the file, `conda-smithy` can tell which version is more recent.
+
+2. An earlier version of this proposal was designed around a migration being a YAML snippet. However, `conda_build_config.yaml` files can have conda-build selectors in them (e.g., `# [osx]`). This conda-build feature makes it pretty hard to use a YAML snippet as a patch on th global pinnings file. Using a direct copy of the proposed new global pinnings file is simpler. This procedure also allows us to migrate almost anything in the file. 
+
+3. Some suggestions have been centered around directly looking at the package versions in the pinnings file to determine which one is newer. This procedure has some flaws. First, it does not allow us to downgrade pinnings if needed. It is ambiguous in the case that we migrate more than one package where one version goes up and the other goes down. Finally, it does not cover changes to things like selectors or zip keys which don't have a version associated with them. 
+
```

Comment:
These alternatives are outdated as the consensus solution is now using a YAML snippet.

Reply:
Would it be possible to label them as such? It might be nice to document paths we considered but did not take (in case anyone asks us later about it)

Reply:
Sure. We also have the other markdown file in the PR that I think is duplicate info at this point.

Reply:
removed